src/ai/backend/common/validators.py

```diff
@@ -354,6 +364,25 @@ def check_and_return(self, value: Any) -> datetime.tzinfo:
         return tz
 
 
+class TimeDuration(t.Trafaret):
+
+    def check_and_return(self, value: Any) -> datetime.timedelta:
+        if not isinstance(value, str):
+            self._failure('value must be string', value=value)
+        if value[-1] not in ['d', 'h', 'm']:
+            self._failure('value is not a known time duration', value=value)
+        try:
+            t = int(value[:-1])
+            if value[-1] == 'd':
+                return datetime.timedelta(days=t)
+            elif value[-1] == 'h':
+                return datetime.timedelta(hours=t)
+            elif value[-1] == 'm':
+                return datetime.timedelta(days=t * 30)
+        except ValueError:
+            self._failure(f'invalid literal for int with base 10: {value[:-1]}', value=value)
```

Comment:
Hm.. I have annotated `_failure()` method as "No Return" so that code paths like this is recognized as a *dead end* by mypy. Maybe we need to update our Trafaret stubs....